### PR TITLE
Updating NonAR to 5.0

### PR DESCRIPTION
**Pull request by devanshidesai25**
Changed the model from autoregressive to non-autoregressive decoding by disabling causal masking, feeding zero tokens to the decoder instead of ground truth tokens during training, enabling all amino acid positions to be predicted simultaneously in parallel. Also removed the [:, :-1] slicing in training_step to predict all positions including the stop token.

### CODE REVIEW

File: casanovo/denovo/model.py [hunk 1]
```diff
@@ -1148,51 +412,118 @@ def validation_step(
...
+            precursors=precursors,
+        )
+
+        # Apply N-terminal modification masking
+        B, L, V = logits.shape
+        nterm_idx = self.nterm_idx.to(device)
+        
+        if self.tokenizer.reverse:
+            # N-term mods only allowed at last position
+            if L > 1:
```
Comment (Lilferrit) [inline casanovo/denovo/model.py]: I don't think the logic here is correct. Since peptides can be different lengths you could have a valid n-terminal modification say in position 5 in one peptide in a batch but position 7 at another peptide in the batch. The logic for the n to c decoding case is correct though.

File: casanovo_20251103225417.mztab
Comment (bittremieux) [inline casanovo_20251103225417.mztab:1]: **todo:** Must be removed.

File: casanovo_5_train.sge
Comment (bittremieux) [inline casanovo_5_train.sge:1]: **todo:** Must be removed.

File: casanovo_sanity_check.sge
Comment (bittremieux) [inline casanovo_sanity_check.sge:1]: **todo:** Must be removed.

File: +
Comment (bittremieux) [inline +:1]: **todo:** Must be removed.

**Review by bittremieux — CHANGES_REQUESTED** [review]
